Makefile

```diff
@@ -52,6 +52,10 @@ generate: ## Code generation (requires operator-sdk >= v0.5.0)
 
 .PHONY: build
 build: ## build binaries
+	@# version must be of the form v1.2.3 with optional suffixes -4 and/or -g56789ab
+	@# or the binary will crash when it tries to parse its version.Raw
+	@echo $(VERSION) | grep -qP '^v\d+\.\d+\.\d+(-\d+)?(-g[a-f0-9]{7,})?(-dirty)?$$' || \
```

Comment:
So another option is to do what support-operator does and just ignore the tags:

https://github.com/openshift/support-operator/blob/master/Makefile#L2

which may make sense if you really, really, really don't care about tags (caring about tags is less of an issue these days).  But this sort of check is fine, just potentially fragile.